include/fiction/algorithms/physical_design/hexagonalization.hpp

```diff
@@ -0,0 +1,200 @@
+//
+// Created by simon on 14.04.23.
+//
+
+#ifndef FICTION_HEXAGONALIZATION_HPP
+#define FICTION_HEXAGONALIZATION_HPP
+
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+namespace fiction
+{
+/**
+ * Utility function to transform a Cartesian coordinate into a hexagonal one.
+ *
+ * @param cartesian_x X-coordinate on the Cartesian grid.
+ * @param cartesian_y Y-coordinate on the Cartesian grid.
+ * @param cartesian_layout_height Height of the Cartesian layout.
+ * @param cartesian_z Z-coordinate on the Cartesian grid.
+ *
+ * @return coordinate on the hexagonal grid.
+ */
+template <typename Lyt>
+coordinate<Lyt> to_hex(int cartesian_x, int cartesian_y, int cartesian_layout_height, int cartesian_z)
```

Comment:
I think this function can be made a bit cleaner by having two template types `CartLyt` and `HexLyt`. Then, It only requires two parameters, namely `coordinate<CartLyt>` and `uint64_t` as the layout height, which can both be `const`. It would return a `coordinate<HexLyt>`.